aesara/tensor/math_opt.py

```diff
@@ -1862,6 +1862,71 @@ def local_neg_div_neg(fgraph, node):
                     return [true_div(new_num, denom)]
 
 
+@register_canonicalize
+@local_optimizer([sub])
+def local_sub_neg_to_add(fgraph, node):
+    """
+    x - (-y) -> x + y
+    """
+
+    if node.op == sub:
+        otype = node.outputs[0].dtype
+        minuend, subtrahend = node.inputs
+
+        if subtrahend.owner:
+            if subtrahend.owner.op == neg:
+                pre_neg = subtrahend.owner.inputs[0]
+                new_out = add(minuend, pre_neg)
+                if new_out.dtype != otype:
```

Comment:
Not sure if this casting is needed. It seems subtraction and negation do not change the variable dtypes even when it would make sense (e.g., `at.neg(scalar(dtype="uint64")`)

Reply:
If your reasoning checks out, then we can't come up with any tests that will hit this and we should remove it.